### PR TITLE
adding talisman as supported evm wallet

### PR DESCRIPTION
**Pull request by pedroapfilho**
This PR adds support not only for Talisman, but for any other wallet that provides a provider on the `window` object.

Screenshot 2022-06-09 at 11 44 55

### CODE REVIEW

**Comment by pedroapfilho** [comment]
Hello @impelcrypto, this PR was created with the talisman extension version 1.5, this is the version that will have the talismanEth on the window

**Comment by impelcrypto** [comment]
@pedroapfilho 
I see. It seems like version 1.5 is not available on the chrome web store yet. Please remind me once it is available so I can test it again.

CC: @hoonsubin 

https://chrome.google.com/webstore/detail/talisman-wallet/fijngjgcjhjmmpcmkeiomlglpeiijkld

image

**Comment by pedroapfilho** [comment]
Screenshot 2022-06-20 at 11 53 13

Added a way to check if the wallet is using the latest version @impelcrypto

**Comment by hoonsubin** [comment]
> @pedroapfilho I see. It seems like version 1.5 is not available on the chrome web store yet. Please remind me once it is available so I can test it again.
> 
> CC: @hoonsubin
> 
> https://chrome.google.com/webstore/detail/talisman-wallet/fijngjgcjhjmmpcmkeiomlglpeiijkld
> 
> image

Please look at the notion link I sent you on Slack as well. It has the instructions for installing the preview version

**Review by hoonsubin — APPROVED** [review]
I confirmed that all the major features are working well (connecting with EVM, signing, sending transactions).

I found some small errors like the page requesting the same signature twice (for the lockdrop account), and this graphical glitch in the token balance (I wish I had that much):

Screen Shot 2022-06-20 at 4 42 22 PM

But I haven't encountered any critical errors yet

File: src/hooks/custom-signature/useEthProvider.ts [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+import { SupportWallet } from 'src/config/wallets';
+import { useStore } from 'src/store';
+import { ref, watchEffect, computed } from 'vue';
 import { EthereumProvider } from '../types/CustomSignature';
 
 export function useEthProvider() {
   const ethProvider = ref<EthereumProvider>();
 
+  const store = useStore();
+  const currentWallet = computed(() => store.getters['general/currentWallet']);
```
Comment (impelcrypto) [inline src/hooks/custom-signature/useEthProvider.ts:10]: Adding custom tokens is not working in both MetaMask and Talisman(EVM).
image

File: src/components/assets/Account.vue [hunk 2]
```diff
@@ -69,7 +72,7 @@
         <span>{{ $t('assets.totalBalance') }}</span>
         <q-skeleton v-if="isSkeleton" animation="fade" class="skeleton--md" />
         <span v-else class="text--total-balance">
-          ${{ $n(balUsd + ttlErc20Amount + ttlNativeXcmUsdAmount) }}
+          ${{ $n(balUsd || 0 + ttlErc20Amount + ttlNativeXcmUsdAmount) }}
```
Comment (impelcrypto) [inline src/components/assets/Account.vue:75]: It seems like the total amount hasn't included ERC20 tokens.
image

**Review by impelcrypto — CHANGES_REQUESTED** [review]

File: src/components/assets/Account.vue (hunk 2 above)
Comment (pedroapfilho) [reply src/components/assets/Account.vue:75]: This was a fix for a TS error that was happening on this file. I'll revert

Comment (impelcrypto) [reply src/components/assets/Account.vue:75]: Sure, but I'm not sure the error comes from this line. Please check the total USD amount with an account that has some ERC20 tokens.

File: src/hooks/custom-signature/useEthProvider.ts (hunk 1 above)
Comment (pedroapfilho) [reply src/hooks/custom-signature/useEthProvider.ts:10]: This shouldn't happen, do you have any idea why the `store` here is undefined?

**Comment by impelcrypto** [comment]
It seems like the selected gas fee is not reflected on the Talisman UI (MetaMask is ok).

https://astar.subscan.io/extrinsic/0xadfa058125474eb9783cfe1482f41a548d964570699305857f4992248eee17a9

image

image

**Pull request merged**